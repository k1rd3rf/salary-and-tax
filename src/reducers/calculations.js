import { FORM_CHANGE } from '../actions';
import { calculate, defaultSalaryParameters } from '../calculations';

const initialState = {
  ...calculate(defaultSalaryParameters),
};

export const calculationsReducer = (
  state = initialState,
  action = {},
  { parameters } = {}
) => {
  switch (action.type) {
    case FORM_CHANGE:
      return Object.assign({}, state, {
        ...calculate(parameters),
      });
    default:
      return state;
  }
};

export default calculationsReducer;
