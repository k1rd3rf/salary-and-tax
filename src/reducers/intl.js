import { intlReducer, UPDATE } from 'react-intl-redux';
import { translationConfig } from '../translation';

export default (state = translationConfig, action = {}) => {
  switch (action.type) {
    case UPDATE:
      return Object.assign({}, state, intlReducer(state, action));
    default:
      return state;
  }
};
