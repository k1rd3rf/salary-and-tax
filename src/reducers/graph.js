import { FORM_CHANGE } from '../actions';
import { defaultSalaryParameters } from '../calculations';
import { getChart } from '../calculations/chart';

const initialState = {
  ...getChart(defaultSalaryParameters),
};

export const graphReducer = (
  state = initialState,
  action = {},
  { parameters } = {}
) => {
  switch (action.type) {
    case FORM_CHANGE:
      return Object.assign({}, state, {
        ...getChart(parameters),
      });
    default:
      return state;
  }
};

export default graphReducer;
