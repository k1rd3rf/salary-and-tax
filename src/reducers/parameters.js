import { FORM_CHANGE } from '../actions';
import { defaultSalaryParameters } from '../calculations';

const initialState = {
  ...defaultSalaryParameters,
};

const handleFormChange = (state, action) => {
  const params = Object.assign({}, state, {
    ...action,
  });
  window.location.hash = JSON.stringify(params);
  return params;
};

const parameters = (state = initialState, action = {}) => {
  switch (action.type) {
    case FORM_CHANGE:
      return handleFormChange(state, action);
    default:
      return state;
  }
};

export default parameters;
