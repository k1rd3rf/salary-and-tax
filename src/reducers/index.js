import reduceReducers from 'reduce-reducers';
import { combineReducers } from 'redux';
import { reducer as formReducer } from 'redux-form';
import calculations from './calculations';
import graph from './graph';
import intl from './intl';
import parameters from './parameters';

const calculationReducer = (state, action) => ({
  ...state,
  calculations: calculations(state.calculations, action, state),
});

const graphReducer = (state, action) => ({
  ...state,
  graph: graph(state.graph, action, state),
});

const rootReducer = reduceReducers(
  combineReducers({
    form: formReducer,
    parameters,
    calculations,
    graph,
    intl,
  }),
  graphReducer,
  calculationReducer
);

export default rootReducer;
