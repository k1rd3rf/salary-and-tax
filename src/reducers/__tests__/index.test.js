import reducers from '../index';

describe('reducers', function() {
  it('should have default values that match snapshot', function() {
    expect(reducers()).toMatchSnapshot();
  });
});
