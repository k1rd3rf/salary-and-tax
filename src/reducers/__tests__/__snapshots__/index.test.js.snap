// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reducers should have default values that match snapshot 1`] = `
Object {
  "calculations": Object {
    "afterTotal": 816923,
    "dividends": Object {
      "afterDividendsTax": 0,
      "dividends": 0,
      "dividendsTax": 0,
      "surplus": 0,
      "surplusTax": 0,
      "upwardsAdjustment": 0,
    },
    "salary": Object {
      "companyIncome": 1712700,
      "employerFee": 211648,
      "income": 1318912,
      "pension": 21313,
      "surplus": 0,
      "theCut": 190300,
      "totalCuts": 584088,
      "turnover": 1903000,
      "vacationSavings": 160827,
      "withoutEmployerFee": 1501052,
      "withoutPension": 1318912,
      "withoutSurplus": 1712700,
      "withoutVacationSavings": 1340225,
    },
    "tax": Object {
      "afterTax": 816923,
      "commonIncome": 1221302,
      "income": 1318912,
      "incomeTax": 280899,
      "minimumDeduction": 97610,
      "socialSecurityDeduction": 108150,
      "step1": 964,
      "step2": 11884,
      "step3": 45136,
      "step4": 54956,
      "totalTax": 501989,
    },
  },
  "form": Object {},
  "graph": Object {
    "annotations": Array [
      Object {
        "labelOptions": Object {
          "backgroundColor": "rgba(255,255,255,0.5)",
          "verticalAlign": "top",
          "y": 15,
        },
        "labels": Array [
          Object {
            "point": Object {
              "x": 1020000,
              "xAxis": 0,
              "y": 533151,
              "yAxis": 0,
            },
            "text": "7.1G",
          },
        ],
      },
    ],
    "dividends": Array [
      Array [
        0,
        0,
      ],
      Array [
        10000,
        5345,
      ],
      Array [
        20000,
        10690,
      ],
      Array [
        30000,
        16034,
      ],
      Array [
        40000,
        21379,
      ],
      Array [
        50000,
        26723,
      ],
      Array [
        60000,
        32068,
      ],
      Array [
        70000,
        37412,
      ],
      Array [
        80000,
        42757,
      ],
      Array [
        90000,
        48102,
      ],
      Array [
        100000,
        53446,
      ],
      Array [
        110000,
        58791,
      ],
      Array [
        120000,
        64135,
      ],
      Array [
        130000,
        69480,
      ],
      Array [
        140000,
        74824,
      ],
      Array [
        150000,
        80169,
      ],
      Array [
        160000,
        85514,
      ],
      Array [
        170000,
        90858,
      ],
      Array [
        180000,
        96203,
      ],
      Array [
        190000,
        101547,
      ],
      Array [
        200000,
        106892,
      ],
      Array [
        210000,
        112236,
      ],
      Array [
        220000,
        117581,
      ],
      Array [
        230000,
        122926,
      ],
      Array [
        240000,
        128270,
      ],
      Array [
        250000,
        133615,
      ],
      Array [
        260000,
        138959,
      ],
      Array [
        270000,
        144304,
      ],
      Array [
        280000,
        149648,
      ],
      Array [
        290000,
        154993,
      ],
      Array [
        300000,
        160338,
      ],
      Array [
        310000,
        165682,
      ],
      Array [
        320000,
        171027,
      ],
      Array [
        330000,
        176371,
      ],
      Array [
        340000,
        181716,
      ],
      Array [
        350000,
        187060,
      ],
      Array [
        360000,
        192405,
      ],
      Array [
        370000,
        197750,
      ],
      Array [
        380000,
        203094,
      ],
      Array [
        390000,
        208439,
      ],
      Array [
        400000,
        213783,
      ],
      Array [
        410000,
        219128,
      ],
      Array [
        420000,
        224472,
      ],
      Array [
        430000,
        229817,
      ],
      Array [
        440000,
        235162,
      ],
      Array [
        450000,
        240506,
      ],
      Array [
        460000,
        245851,
      ],
      Array [
        470000,
        251195,
      ],
      Array [
        480000,
        256540,
      ],
      Array [
        490000,
        261884,
      ],
      Array [
        500000,
        267229,
      ],
      Array [
        510000,
        272574,
      ],
      Array [
        520000,
        277918,
      ],
      Array [
        530000,
        283263,
      ],
      Array [
        540000,
        288607,
      ],
      Array [
        550000,
        293952,
      ],
      Array [
        560000,
        299296,
      ],
      Array [
        570000,
        304641,
      ],
      Array [
        580000,
        309986,
      ],
      Array [
        590000,
        315330,
      ],
      Array [
        600000,
        320675,
      ],
      Array [
        610000,
        326019,
      ],
      Array [
        620000,
        331364,
      ],
      Array [
        630000,
        336708,
      ],
      Array [
        640000,
        342053,
      ],
      Array [
        650000,
        347398,
      ],
      Array [
        660000,
        352742,
      ],
      Array [
        670000,
        358087,
      ],
      Array [
        680000,
        363431,
      ],
      Array [
        690000,
        368776,
      ],
      Array [
        700000,
        374120,
      ],
      Array [
        710000,
        379465,
      ],
      Array [
        720000,
        384810,
      ],
      Array [
        730000,
        390154,
      ],
      Array [
        740000,
        395499,
      ],
      Array [
        750000,
        400843,
      ],
      Array [
        760000,
        406188,
      ],
      Array [
        770000,
        411532,
      ],
      Array [
        780000,
        416877,
      ],
      Array [
        790000,
        422222,
      ],
      Array [
        800000,
        427566,
      ],
      Array [
        810000,
        432911,
      ],
      Array [
        820000,
        438255,
      ],
      Array [
        830000,
        443600,
      ],
      Array [
        840000,
        448944,
      ],
      Array [
        850000,
        454289,
      ],
      Array [
        860000,
        459634,
      ],
      Array [
        870000,
        464978,
      ],
      Array [
        880000,
        470323,
      ],
      Array [
        890000,
        475667,
      ],
      Array [
        900000,
        481012,
      ],
      Array [
        910000,
        486356,
      ],
      Array [
        920000,
        491701,
      ],
      Array [
        930000,
        497046,
      ],
      Array [
        940000,
        502390,
      ],
      Array [
        950000,
        507735,
      ],
      Array [
        960000,
        513079,
      ],
      Array [
        970000,
        518424,
      ],
      Array [
        980000,
        523768,
      ],
      Array [
        990000,
        529113,
      ],
      Array [
        1000000,
        534457,
      ],
      Array [
        1010000,
        539802,
      ],
      Array [
        1020000,
        545147,
      ],
      Array [
        1030000,
        550491,
      ],
      Array [
        1040000,
        555836,
      ],
      Array [
        1050000,
        561180,
      ],
      Array [
        1060000,
        566525,
      ],
      Array [
        1070000,
        571869,
      ],
      Array [
        1080000,
        577214,
      ],
      Array [
        1090000,
        582559,
      ],
      Array [
        1100000,
        587903,
      ],
      Array [
        1110000,
        593248,
      ],
      Array [
        1120000,
        598592,
      ],
      Array [
        1130000,
        603937,
      ],
      Array [
        1140000,
        609281,
      ],
      Array [
        1150000,
        614626,
      ],
      Array [
        1160000,
        619971,
      ],
      Array [
        1170000,
        625315,
      ],
      Array [
        1180000,
        630660,
      ],
      Array [
        1190000,
        636004,
      ],
      Array [
        1200000,
        641349,
      ],
      Array [
        1210000,
        646693,
      ],
      Array [
        1220000,
        652038,
      ],
      Array [
        1230000,
        657383,
      ],
      Array [
        1240000,
        662727,
      ],
      Array [
        1250000,
        668072,
      ],
      Array [
        1260000,
        673416,
      ],
      Array [
        1270000,
        678761,
      ],
      Array [
        1280000,
        684105,
      ],
      Array [
        1290000,
        689450,
      ],
      Array [
        1300000,
        694795,
      ],
      Array [
        1310000,
        700139,
      ],
      Array [
        1320000,
        705484,
      ],
      Array [
        1330000,
        710828,
      ],
      Array [
        1340000,
        716173,
      ],
      Array [
        1350000,
        721517,
      ],
      Array [
        1360000,
        726862,
      ],
      Array [
        1370000,
        732207,
      ],
      Array [
        1380000,
        737551,
      ],
      Array [
        1390000,
        742896,
      ],
      Array [
        1400000,
        748240,
      ],
      Array [
        1410000,
        753585,
      ],
      Array [
        1420000,
        758929,
      ],
      Array [
        1430000,
        764274,
      ],
      Array [
        1440000,
        769619,
      ],
      Array [
        1450000,
        774963,
      ],
      Array [
        1460000,
        780308,
      ],
      Array [
        1470000,
        785652,
      ],
      Array [
        1480000,
        790997,
      ],
      Array [
        1490000,
        796341,
      ],
      Array [
        1500000,
        801686,
      ],
      Array [
        1510000,
        807031,
      ],
      Array [
        1520000,
        812375,
      ],
      Array [
        1530000,
        817720,
      ],
      Array [
        1540000,
        823064,
      ],
      Array [
        1550000,
        828409,
      ],
      Array [
        1560000,
        833753,
      ],
      Array [
        1570000,
        839098,
      ],
      Array [
        1580000,
        844443,
      ],
      Array [
        1590000,
        849787,
      ],
      Array [
        1600000,
        855132,
      ],
      Array [
        1610000,
        860476,
      ],
      Array [
        1620000,
        865821,
      ],
      Array [
        1630000,
        871165,
      ],
      Array [
        1640000,
        876510,
      ],
      Array [
        1650000,
        881855,
      ],
      Array [
        1660000,
        887199,
      ],
      Array [
        1670000,
        892544,
      ],
      Array [
        1680000,
        897888,
      ],
      Array [
        1690000,
        903233,
      ],
      Array [
        1700000,
        908577,
      ],
    ],
    "income": Array [
      Array [
        0,
        1318912,
      ],
      Array [
        10000,
        1311087,
      ],
      Array [
        20000,
        1303262,
      ],
      Array [
        30000,
        1295437,
      ],
      Array [
        40000,
        1287612,
      ],
      Array [
        50000,
        1279787,
      ],
      Array [
        60000,
        1271962,
      ],
      Array [
        70000,
        1264137,
      ],
      Array [
        80000,
        1256311,
      ],
      Array [
        90000,
        1248486,
      ],
      Array [
        100000,
        1240661,
      ],
      Array [
        110000,
        1232836,
      ],
      Array [
        120000,
        1225010,
      ],
      Array [
        130000,
        1217185,
      ],
      Array [
        140000,
        1209360,
      ],
      Array [
        150000,
        1201535,
      ],
      Array [
        160000,
        1193709,
      ],
      Array [
        170000,
        1185884,
      ],
      Array [
        180000,
        1178059,
      ],
      Array [
        190000,
        1170234,
      ],
      Array [
        200000,
        1162408,
      ],
      Array [
        210000,
        1154583,
      ],
      Array [
        220000,
        1146758,
      ],
      Array [
        230000,
        1138980,
      ],
      Array [
        240000,
        1131310,
      ],
      Array [
        250000,
        1123642,
      ],
      Array [
        260000,
        1115973,
      ],
      Array [
        270000,
        1108305,
      ],
      Array [
        280000,
        1100635,
      ],
      Array [
        290000,
        1092967,
      ],
      Array [
        300000,
        1085298,
      ],
      Array [
        310000,
        1077630,
      ],
      Array [
        320000,
        1069961,
      ],
      Array [
        330000,
        1062292,
      ],
      Array [
        340000,
        1054624,
      ],
      Array [
        350000,
        1046955,
      ],
      Array [
        360000,
        1039287,
      ],
      Array [
        370000,
        1031617,
      ],
      Array [
        380000,
        1023949,
      ],
      Array [
        390000,
        1016280,
      ],
      Array [
        400000,
        1008612,
      ],
      Array [
        410000,
        1000943,
      ],
      Array [
        420000,
        993275,
      ],
      Array [
        430000,
        985606,
      ],
      Array [
        440000,
        977938,
      ],
      Array [
        450000,
        970268,
      ],
      Array [
        460000,
        962600,
      ],
      Array [
        470000,
        954931,
      ],
      Array [
        480000,
        947263,
      ],
      Array [
        490000,
        939593,
      ],
      Array [
        500000,
        931925,
      ],
      Array [
        510000,
        924256,
      ],
      Array [
        520000,
        916588,
      ],
      Array [
        530000,
        908918,
      ],
      Array [
        540000,
        901250,
      ],
      Array [
        550000,
        893581,
      ],
      Array [
        560000,
        885913,
      ],
      Array [
        570000,
        878243,
      ],
      Array [
        580000,
        870575,
      ],
      Array [
        590000,
        862906,
      ],
      Array [
        600000,
        855238,
      ],
      Array [
        610000,
        847568,
      ],
      Array [
        620000,
        839900,
      ],
      Array [
        630000,
        832231,
      ],
      Array [
        640000,
        824563,
      ],
      Array [
        650000,
        816893,
      ],
      Array [
        660000,
        809225,
      ],
      Array [
        670000,
        801556,
      ],
      Array [
        680000,
        793888,
      ],
      Array [
        690000,
        786219,
      ],
      Array [
        700000,
        778550,
      ],
      Array [
        710000,
        770881,
      ],
      Array [
        720000,
        763213,
      ],
      Array [
        730000,
        755544,
      ],
      Array [
        740000,
        747875,
      ],
      Array [
        750000,
        740206,
      ],
      Array [
        760000,
        732538,
      ],
      Array [
        770000,
        724869,
      ],
      Array [
        780000,
        717201,
      ],
      Array [
        790000,
        709532,
      ],
      Array [
        800000,
        701864,
      ],
      Array [
        810000,
        694195,
      ],
      Array [
        820000,
        686526,
      ],
      Array [
        830000,
        678857,
      ],
      Array [
        840000,
        671188,
      ],
      Array [
        850000,
        663519,
      ],
      Array [
        860000,
        655850,
      ],
      Array [
        870000,
        648182,
      ],
      Array [
        880000,
        640513,
      ],
      Array [
        890000,
        632845,
      ],
      Array [
        900000,
        625175,
      ],
      Array [
        910000,
        617507,
      ],
      Array [
        920000,
        609838,
      ],
      Array [
        930000,
        602170,
      ],
      Array [
        940000,
        594500,
      ],
      Array [
        950000,
        586832,
      ],
      Array [
        960000,
        579163,
      ],
      Array [
        970000,
        571495,
      ],
      Array [
        980000,
        563826,
      ],
      Array [
        990000,
        556157,
      ],
      Array [
        1000000,
        548488,
      ],
      Array [
        1010000,
        540820,
      ],
      Array [
        1020000,
        533151,
      ],
      Array [
        1030000,
        525482,
      ],
      Array [
        1040000,
        517813,
      ],
      Array [
        1050000,
        510145,
      ],
      Array [
        1060000,
        502476,
      ],
      Array [
        1070000,
        494807,
      ],
      Array [
        1080000,
        487138,
      ],
      Array [
        1090000,
        479470,
      ],
      Array [
        1100000,
        471801,
      ],
      Array [
        1110000,
        464132,
      ],
      Array [
        1120000,
        456463,
      ],
      Array [
        1130000,
        448795,
      ],
      Array [
        1140000,
        441126,
      ],
      Array [
        1150000,
        433457,
      ],
      Array [
        1160000,
        425788,
      ],
      Array [
        1170000,
        418120,
      ],
      Array [
        1180000,
        410451,
      ],
      Array [
        1190000,
        402783,
      ],
      Array [
        1200000,
        395114,
      ],
      Array [
        1210000,
        387446,
      ],
      Array [
        1220000,
        379777,
      ],
      Array [
        1230000,
        372108,
      ],
      Array [
        1240000,
        364439,
      ],
      Array [
        1250000,
        356771,
      ],
      Array [
        1260000,
        349102,
      ],
      Array [
        1270000,
        341433,
      ],
      Array [
        1280000,
        333764,
      ],
      Array [
        1290000,
        326096,
      ],
      Array [
        1300000,
        318427,
      ],
      Array [
        1310000,
        310759,
      ],
      Array [
        1320000,
        303089,
      ],
      Array [
        1330000,
        295421,
      ],
      Array [
        1340000,
        287752,
      ],
      Array [
        1350000,
        280084,
      ],
      Array [
        1360000,
        272414,
      ],
      Array [
        1370000,
        264746,
      ],
      Array [
        1380000,
        257077,
      ],
      Array [
        1390000,
        249409,
      ],
      Array [
        1400000,
        241739,
      ],
      Array [
        1410000,
        234071,
      ],
      Array [
        1420000,
        226402,
      ],
      Array [
        1430000,
        218734,
      ],
      Array [
        1440000,
        211064,
      ],
      Array [
        1450000,
        203396,
      ],
      Array [
        1460000,
        195727,
      ],
      Array [
        1470000,
        188059,
      ],
      Array [
        1480000,
        180389,
      ],
      Array [
        1490000,
        172721,
      ],
      Array [
        1500000,
        165052,
      ],
      Array [
        1510000,
        157384,
      ],
      Array [
        1520000,
        149714,
      ],
      Array [
        1530000,
        142046,
      ],
      Array [
        1540000,
        134377,
      ],
      Array [
        1550000,
        126709,
      ],
      Array [
        1560000,
        119040,
      ],
      Array [
        1570000,
        111372,
      ],
      Array [
        1580000,
        103703,
      ],
      Array [
        1590000,
        96017,
      ],
      Array [
        1600000,
        88192,
      ],
      Array [
        1610000,
        80366,
      ],
      Array [
        1620000,
        72541,
      ],
      Array [
        1630000,
        64716,
      ],
      Array [
        1640000,
        56891,
      ],
      Array [
        1650000,
        49065,
      ],
      Array [
        1660000,
        41240,
      ],
      Array [
        1670000,
        33415,
      ],
      Array [
        1680000,
        25590,
      ],
      Array [
        1690000,
        17764,
      ],
      Array [
        1700000,
        9939,
      ],
    ],
    "total": Array [
      Array [
        0,
        816923,
      ],
      Array [
        10000,
        818089,
      ],
      Array [
        20000,
        819256,
      ],
      Array [
        30000,
        820421,
      ],
      Array [
        40000,
        821587,
      ],
      Array [
        50000,
        822753,
      ],
      Array [
        60000,
        823920,
      ],
      Array [
        70000,
        825084,
      ],
      Array [
        80000,
        826250,
      ],
      Array [
        90000,
        827417,
      ],
      Array [
        100000,
        828582,
      ],
      Array [
        110000,
        829749,
      ],
      Array [
        120000,
        830914,
      ],
      Array [
        130000,
        832080,
      ],
      Array [
        140000,
        833246,
      ],
      Array [
        150000,
        834413,
      ],
      Array [
        160000,
        835578,
      ],
      Array [
        170000,
        836743,
      ],
      Array [
        180000,
        837910,
      ],
      Array [
        190000,
        839075,
      ],
      Array [
        200000,
        840241,
      ],
      Array [
        210000,
        841407,
      ],
      Array [
        220000,
        842572,
      ],
      Array [
        230000,
        843764,
      ],
      Array [
        240000,
        845012,
      ],
      Array [
        250000,
        846263,
      ],
      Array [
        260000,
        847512,
      ],
      Array [
        270000,
        848762,
      ],
      Array [
        280000,
        850010,
      ],
      Array [
        290000,
        851260,
      ],
      Array [
        300000,
        852510,
      ],
      Array [
        310000,
        853760,
      ],
      Array [
        320000,
        855010,
      ],
      Array [
        330000,
        856259,
      ],
      Array [
        340000,
        857508,
      ],
      Array [
        350000,
        858757,
      ],
      Array [
        360000,
        860008,
      ],
      Array [
        370000,
        861257,
      ],
      Array [
        380000,
        862507,
      ],
      Array [
        390000,
        863756,
      ],
      Array [
        400000,
        865005,
      ],
      Array [
        410000,
        866255,
      ],
      Array [
        420000,
        867505,
      ],
      Array [
        430000,
        868754,
      ],
      Array [
        440000,
        870005,
      ],
      Array [
        450000,
        871253,
      ],
      Array [
        460000,
        872503,
      ],
      Array [
        470000,
        873538,
      ],
      Array [
        480000,
        874558,
      ],
      Array [
        490000,
        875576,
      ],
      Array [
        500000,
        876597,
      ],
      Array [
        510000,
        877617,
      ],
      Array [
        520000,
        878636,
      ],
      Array [
        530000,
        879655,
      ],
      Array [
        540000,
        880674,
      ],
      Array [
        550000,
        881694,
      ],
      Array [
        560000,
        882713,
      ],
      Array [
        570000,
        883733,
      ],
      Array [
        580000,
        884752,
      ],
      Array [
        590000,
        885770,
      ],
      Array [
        600000,
        886791,
      ],
      Array [
        610000,
        887809,
      ],
      Array [
        620000,
        888830,
      ],
      Array [
        630000,
        889849,
      ],
      Array [
        640000,
        890868,
      ],
      Array [
        650000,
        891887,
      ],
      Array [
        660000,
        892907,
      ],
      Array [
        670000,
        893927,
      ],
      Array [
        680000,
        894947,
      ],
      Array [
        690000,
        895966,
      ],
      Array [
        700000,
        896983,
      ],
      Array [
        710000,
        898003,
      ],
      Array [
        720000,
        899024,
      ],
      Array [
        730000,
        900043,
      ],
      Array [
        740000,
        901063,
      ],
      Array [
        750000,
        902081,
      ],
      Array [
        760000,
        903101,
      ],
      Array [
        770000,
        904120,
      ],
      Array [
        780000,
        905141,
      ],
      Array [
        790000,
        906160,
      ],
      Array [
        800000,
        907180,
      ],
      Array [
        810000,
        908200,
      ],
      Array [
        820000,
        909217,
      ],
      Array [
        830000,
        910237,
      ],
      Array [
        840000,
        911256,
      ],
      Array [
        850000,
        912275,
      ],
      Array [
        860000,
        913295,
      ],
      Array [
        870000,
        914315,
      ],
      Array [
        880000,
        915334,
      ],
      Array [
        890000,
        916353,
      ],
      Array [
        900000,
        917373,
      ],
      Array [
        910000,
        918392,
      ],
      Array [
        920000,
        919412,
      ],
      Array [
        930000,
        920433,
      ],
      Array [
        940000,
        921126,
      ],
      Array [
        950000,
        921448,
      ],
      Array [
        960000,
        921769,
      ],
      Array [
        970000,
        922092,
      ],
      Array [
        980000,
        922413,
      ],
      Array [
        990000,
        922735,
      ],
      Array [
        1000000,
        923055,
      ],
      Array [
        1010000,
        923377,
      ],
      Array [
        1020000,
        923699,
      ],
      Array [
        1030000,
        924020,
      ],
      Array [
        1040000,
        924342,
      ],
      Array [
        1050000,
        924663,
      ],
      Array [
        1060000,
        924984,
      ],
      Array [
        1070000,
        925306,
      ],
      Array [
        1080000,
        925628,
      ],
      Array [
        1090000,
        925951,
      ],
      Array [
        1100000,
        926272,
      ],
      Array [
        1110000,
        926593,
      ],
      Array [
        1120000,
        926914,
      ],
      Array [
        1130000,
        927236,
      ],
      Array [
        1140000,
        927557,
      ],
      Array [
        1150000,
        927879,
      ],
      Array [
        1160000,
        928201,
      ],
      Array [
        1170000,
        928522,
      ],
      Array [
        1180000,
        928844,
      ],
      Array [
        1190000,
        929165,
      ],
      Array [
        1200000,
        929487,
      ],
      Array [
        1210000,
        929808,
      ],
      Array [
        1220000,
        930131,
      ],
      Array [
        1230000,
        930453,
      ],
      Array [
        1240000,
        930774,
      ],
      Array [
        1250000,
        931095,
      ],
      Array [
        1260000,
        931416,
      ],
      Array [
        1270000,
        931738,
      ],
      Array [
        1280000,
        932059,
      ],
      Array [
        1290000,
        932382,
      ],
      Array [
        1300000,
        932703,
      ],
      Array [
        1310000,
        933024,
      ],
      Array [
        1320000,
        933345,
      ],
      Array [
        1330000,
        933667,
      ],
      Array [
        1340000,
        933989,
      ],
      Array [
        1350000,
        934310,
      ],
      Array [
        1360000,
        934633,
      ],
      Array [
        1370000,
        934954,
      ],
      Array [
        1380000,
        935275,
      ],
      Array [
        1390000,
        935598,
      ],
      Array [
        1400000,
        935918,
      ],
      Array [
        1410000,
        936167,
      ],
      Array [
        1420000,
        936342,
      ],
      Array [
        1430000,
        936518,
      ],
      Array [
        1440000,
        936089,
      ],
      Array [
        1450000,
        935471,
      ],
      Array [
        1460000,
        934853,
      ],
      Array [
        1470000,
        934236,
      ],
      Array [
        1480000,
        933617,
      ],
      Array [
        1490000,
        932998,
      ],
      Array [
        1500000,
        932325,
      ],
      Array [
        1510000,
        931601,
      ],
      Array [
        1520000,
        930875,
      ],
      Array [
        1530000,
        930151,
      ],
      Array [
        1540000,
        929425,
      ],
      Array [
        1550000,
        928700,
      ],
      Array [
        1560000,
        927974,
      ],
      Array [
        1570000,
        927250,
      ],
      Array [
        1580000,
        926525,
      ],
      Array [
        1590000,
        925785,
      ],
      Array [
        1600000,
        924937,
      ],
      Array [
        1610000,
        924086,
      ],
      Array [
        1620000,
        923238,
      ],
      Array [
        1630000,
        922389,
      ],
      Array [
        1640000,
        921540,
      ],
      Array [
        1650000,
        920691,
      ],
      Array [
        1660000,
        919842,
      ],
      Array [
        1670000,
        918992,
      ],
      Array [
        1680000,
        918143,
      ],
      Array [
        1690000,
        917294,
      ],
      Array [
        1700000,
        916445,
      ],
    ],
  },
  "intl": Object {
    "locale": "nb",
    "locales": Array [
      Object {
        "active": true,
        "label": "Norsk",
        "value": "nb",
      },
      Object {
        "active": false,
        "label": "English",
        "value": "en",
      },
    ],
    "messages": Object {
      "salary.calculations.employer.fee": "Arbeidsgiveravgift",
      "salary.calculations.header": "Inntekstberegninger",
      "salary.calculations.income": "Brutto inntekt",
      "salary.calculations.pension": "Pensjon",
      "salary.calculations.subcontractor.cut": "Andel til overleverandør",
      "salary.calculations.surplus": "Overskudd",
      "salary.calculations.turnover": "Omsetning",
      "salary.calculations.vacation.savings": "Feriepenger",
      "salary.parameters.employer.fee": "Arbeidsgiveravgift",
      "salary.parameters.header": "Parametere",
      "salary.parameters.hour.rate": "Timepris",
      "salary.parameters.hours.per.year": "Timer per år",
      "salary.parameters.include.in.calculation": "Inkluder i utregningen",
      "salary.parameters.locale": "Språk",
      "salary.parameters.pension": "Pensjon",
      "salary.parameters.pension.step1": "(%) (1G-7.1G)",
      "salary.parameters.pension.step2": "(%) (7.1G-12G)",
      "salary.parameters.percentage": "(%)",
      "salary.parameters.subcontractor": "Overleverandør",
      "salary.parameters.subcontractor.cut": "Andel (%)",
      "salary.parameters.submit": "Beregn",
      "salary.parameters.surplus": "Overskudd",
      "salary.parameters.surplus.amount": "Beløp",
      "salary.parameters.turnover": "Omsetning",
      "salary.parameters.vacation.savings": "Feriepenger",
      "summary.dividends": "Utbytte",
      "summary.header": "Sammendrag",
      "summary.income": "Netto inntekt første år",
      "summary.total.cuts": "Andel til overleverandør, overskudd, arbeidsgiveravgift, feriepenger, pensjon",
      "summary.total.income": "Total inntjening første år (uten feriepenger)",
      "summary.total.tax": "Skatt sammenlagt",
      "summary.turnover": "Omsetning",
      "surplus.graph.chart.header": "Inntekt vs. overskudd",
      "surplus.graph.header": "Inntekt vs. overskudd",
      "surplus.graph.x.axis.title": "Overskudd",
      "surplus.graph.y.axis.title": "NOK",
      "surplus.series.dividends": "Utbytte",
      "surplus.series.income": "Netto inntekt",
      "surplus.series.total": "Sammenlagt",
      "surplus.suffix": "NOK",
      "tax.calculations.after.tax": "Netto inntekt",
      "tax.calculations.common.income": "Inntektsskatt-grunnlag",
      "tax.calculations.common.income.header": "Inntektsskatt-grunnlag",
      "tax.calculations.dividends.amount": "Utbytte",
      "tax.calculations.dividends.tax": "Skatt på utbytte",
      "tax.calculations.header": "Skatteberegninger",
      "tax.calculations.income": "Brutto inntekt",
      "tax.calculations.income.header": "Inntekt",
      "tax.calculations.income.tax": "Inntektsskatt (23%)",
      "tax.calculations.minimum.deduction": "Minstefradrag (45%)",
      "tax.calculations.social.security.deduction": "Trygdeavgift (8.2%)",
      "tax.calculations.step.1": "Trinnskatt trinn 1 (1.4%)",
      "tax.calculations.step.2": "Trinnskatt trinn 2 (3.3%)",
      "tax.calculations.step.3": "Trinnskatt trinn 3 (12.4%)",
      "tax.calculations.step.4": "Trinnskatt trinn 4 (15.4%)",
      "tax.calculations.surplus.amount": "Selskapets overskudd",
      "tax.calculations.surplus.header": "Overskudd",
      "tax.calculations.surplus.tax": "Skatt på overskudd (23%)",
    },
  },
  "parameters": Object {
    "cut": 10,
    "employerFee": Object {
      "include": true,
      "rate": 14.1,
    },
    "ensuringFee": 15000,
    "graph": Object {
      "increments": 10000,
    },
    "hourRate": 1100,
    "hoursPerYear": 1730,
    "locale": "nb",
    "pension": Object {
      "include": true,
      "step1": Object {
        "rate": 2,
        "threshold": Object {
          "high": 687869,
          "low": 96883,
        },
      },
      "step2": Object {
        "rate": 2,
        "threshold": Object {
          "high": 1162596,
          "low": 687869,
        },
      },
    },
    "surplus": Object {
      "amount": 0,
      "include": true,
    },
    "vacationSavings": Object {
      "include": true,
      "rate": 12,
    },
  },
}
`;
